chaco/polygon_plot.py

```diff
@@ -134,11 +144,23 @@ def hittest(self, screen_pt, threshold=7.0, return_distance=False):
         data_pt = self.map_data(screen_pt, all_values=True)
         index = self.index.get_data()
         value = self.value.get_data()
-        poly = np.vstack((index,value)).T
-        if points_in_polygon([data_pt], poly)[0] == 1:
-            return True
-        else:
-            return False
+        points = zip(index,value)
```

Comment:
Avoid using `zip` with `numpy` arrays (there are, unfortunately too many places where this already happens in the Chaco code base), and instead use things like `np.vstack` or `np.column_stack` which are vectorized and return numpy arrays.